### PR TITLE
Refactoring tests. Remove dev dependencies from league and yiisoft containers

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Awesome. That removes circular dependency when it's installed in dev mode.

File: tests/Unit/FactoryTest.php [hunk 1]
```diff
@@ -266,4 +304,48 @@ public function testDoNotResolveDependenciesFromFactory(): void
...
+    {
+        $container = new SimpleContainer(
+            [],
+            static function (string $id) use (&$container) {
+                if ($id === ContainerInterface::class) {
+                    return $container;
+                }
+                throw new NotFoundException($id);
+            }
+        );
```
Comment (roxblnfk) [inline tests/Unit/FactoryTest.php:318]:
```suggestion
        $container = new SimpleContainer(
            [ContainerInterface::class => &$container]
        );
```

File: tests/Unit/FactoryTest.php [hunk 2]
```diff
@@ -250,9 +281,16 @@ public function testDoNotFallbackToContainer(): void
...
-        ]);
+        $container = new SimpleContainer(
+            [],
+            static function (string $id) {
+                if ($id === EngineInterface::class) {
+                    return new EngineMarkOne();
+                }
+                throw new NotFoundException($id);
+            }
+        );
```
Comment (roxblnfk) [inline tests/Unit/FactoryTest.php:292]:
```suggestion
        $container = new SimpleContainer(
            [EngineInterface::class => new EngineMarkOne()]
        );
```

File: tests/Unit/FactoryTest.php [hunk 3]
```diff
@@ -231,12 +257,17 @@ public function testResolveDependenciesUsingContainer(): void
...
+            [],
+            static function (string $id) {
+                if ($id === EngineMarkOne::class) {
+                    $engine = new EngineMarkOne();
+                    $engine->setNumber(42);
+                    return $engine;
+                }
+                throw new NotFoundException($id);
+            }
+        );
```
Comment (roxblnfk) [inline tests/Unit/FactoryTest.php:270]:
```suggestion
        $engine = new EngineMarkOne();
        $engine->setNumber(42);
        $container = new SimpleContainer(
            [EngineMarkOne::class => $engine]
        );
```

File: tests/Unit/Definitions/DynamicReferenceTest.php [hunk 4]
```diff
@@ -6,18 +6,32 @@
...
+            static function (string $id) use (&$container) {
+                if ($id === ContainerInterface::class) {
+                    return $container;
+                }
+                if ($id === Injector::class) {
+                    return new Injector($container);
+                }
+                throw new NotFoundException($id);
+            }
+        );
```
Comment (roxblnfk) [inline tests/Unit/Definitions/DynamicReferenceTest.php:33]:
```suggestion
        $container = new SimpleContainer(
            [
                ContainerInterface::class => &$container,
                EngineInterface::class => new EngineMarkOne(),
            ],
            static function (string $id) use (&$container) {
                if ($id === Injector::class) {
                    return new Injector($container);
                }
                throw new NotFoundException($id);
            }
        );
```

**Review by roxblnfk — COMMENTED** [review]
SimpleContainer can be used more simple

**Comment by samdark** [comment]
@roxblnfk would you please do a pull request with these changes? Thanks.

**Pull request merged**